code/datums/mob_stagehandler.dm

```diff
@@ -0,0 +1,44 @@
+//Meant to handle stages of transitions or effects on mobs instead of humans having like 8 vars defined just for SCPs
+/datum/stageHandler
+	var/list/stages
+
+/datum/stageHandler/New()
+	. = ..()
+	LAZYINITLIST(stages)
+
+///Creates a new stage to track
+/datum/stageHandler/proc/createStage(stageID)
+	if(stages[stageID])
```

Comment:
It won't trigger on stage being equal to 0, so you need to check for null. Intended?